modules/apps/change-tracking/change-tracking-web/src/main/resources/META-INF/resources/publications/js/views/ChangeTrackingRenderView.js

```diff
@@ -1490,7 +1489,7 @@ export default function ChangeTrackingRenderView({
 				newDataURL =
 					initialDataURL +
 					'&' +
-					namespace +
+					'_com_liferay_change_tracking_web_portlet_PublicationsPortlet_' +
```

Comment:
@ctang3 I think we can simplify this block of code for setting the `segmentsExperienceId` with createPortletURL(baseURL, params={}) from `frontend-js-web`. It will override the existing value or create a new one for the given param.

Reply:
Thanks for pointing out that createPortletURL can do this same logic, I'll remember that in the future. 

I made the change to replace the `newDataURL` logic, but for some reason using createPortletURL() instead of that logic is causing an infinite re-render loop in the ExperienceDropdown component, which has a `useEffect()` that calls this `updatePreviewRender()`. I'm working through that now, but is this something you've encountered/recognized before?

Reply:
Pushed the changes. Removing updatePreviewRenderer from the useEffect dependencies resolved the infinite re-render loop. I'm not sure what about createPortletURL makes it think the value is always changing though.

Reply:
I'm wondering if it has to do with the return object. `createPortletURL` actually returns an url object and dataURL is expected to be a string. Can you check if `setDataURL(newDataURL.toString())` fix the issue?

Reply:
Ahh mystery solved! Looks like that was the reason - calling `toString()` also fixed the issue. 

I switched to that and squashed the `toString()` change into my previous commit. I think this is better since if I'm not mistaken, its best practice that updatePreviewRenderer _should_ be in the useEffect dependencies.